src/sync/sync_config.hpp

```diff
@@ -110,14 +112,40 @@ struct SyncConfig {
     util::Optional<std::string> ssl_trust_certificate_path;
     std::function<sync::Session::SSLVerifyCallback> ssl_verify_callback;
     bool validate_sync_history = true;
+    bool is_partial = false;
+    util::Optional<std::string> custom_partial_sync_identifier;
+
+    std::string resolved_realm_url() const
+    {
+        REALM_ASSERT(realm_url.length() > 0);
+        REALM_ASSERT(user);
+        std::string base_url = realm_url;
+        if (base_url.back() == '/')
```

Comment:
I haven't tried a Realm URL with a trailing slash, but I think they should be rejected (since such a path would indicate a directory path). It may make more sense to throw an exception or take a similar action if the path has a trailing slash, rather than silently stripping it. (On the other hand, I don't want a path to be constructed with two consecutive slashes and then have it fail much later in the sync pipeline.)